projects/igniteui-angular/src/lib/grids/common/crud.service.ts

```diff
@@ -251,6 +251,11 @@ export class IgxCellCrudState {
         if (!this.cell) {
             return;
         }
+        // this is needed when we are not using ngModel to update the editValue
+        // so that the change event of the inlineEditorTemplate is hit before
+        // trying to update any cell
+        const activeElement = document.activeElement as HTMLElement;
```

Comment:
Interestingly, because I noticed the formControl/validation code just below, could this interact with that? Considering it's for the blur trigger and patched more/less the lack of blur before (almost like we've hit this before?🙃), this could make the validation patch redundant?
There's still a slight chance the formControl updates on the next cycle... `Promise.resolve` if I recall correctly, so perhaps still needed. Regardless, should test carefully that interaction with the fix.

Reply:
About the validation: from what I see is that it works correct. Also, the code you are saying below the fix would be always used if the validationTrigger is blur. I dont think that we can remove the logic without further refactoring because if we remove that logic (the `cell.pendingValue` which is set when the editValue is changed but formControl is updated right before the grid tbody is focused) then the validation works like it is always updating on "change".